refinery/core/test_views.py

```diff
@@ -934,3 +939,86 @@ def test_get_workflow_graph(self):
             uuid=self.workflow.uuid
         )
         self.assertEqual(get_response.content, self.mock_workflow_graph)
+
+
+class EventApiV2Tests(APIV2TestCase):
+    def setUp(self):
+        create_public_group()
+        # Prints a warning and continues if public group already exists.
+        # Necessary if using "--settings=config.settings.quick_test".
+        # TODO: Move to superclass?
+
+        super(EventApiV2Tests, self).setUp(
+            api_base_name="events/",
+            view=EventViewSet.as_view({"get": "list"})
+        )
+
+    def test_get_event_list(self):
+        user = User.objects.create_user('testuser')
+        CuserMiddleware.set_user(user)
+
+        create_tool_with_necessary_models("VISUALIZATION")
+        create_tool_with_necessary_models("WORKFLOW")
+
+        events = Event.objects.all()
+        self.assertEqual(len(events), 4)
+
+        messages = [str(_) for _ in events]
+        data_sets = [_.data_set.uuid for _ in events]
+        display_names = [
+            json.loads(_.json).get('display_name') for _ in events
+        ]
+        date_times = [
+            _.date_time.isoformat().replace('+00:00', 'Z') for _ in events
+        ]
+
+        get_request = self.factory.get(urljoin(self.url_root, '/'))
+        get_response = self.view(get_request).render()
+        # TODO: Why do I need render()?
```

Comment:
In general, can we remove todo questions from the code and add them to the dev meeting agenda?